airflow/auth/managers/base_auth_manager.py

```diff
@@ -102,21 +102,21 @@ def deserialize_user(self, token: dict[str, Any]) -> T:
         """Create a user object from dict."""
 
     @abstractmethod
-    def serialize_user(self, user: T) -> dict[str, Any]:
-        """Create a dict from a user object."""
+    def serialize_user(self, user: T) -> tuple[str, dict[str, Any]]:
+        """Create a subject and extra claims dict from a user object."""
```

Comment:
I think the `subject` should be part of the User interface. And we shouldn't use a `serialize_user` method to extract the appropriate user subject.

Reply:
Could the auth api use an anonymous user ? or a technical user ? Anyway if the current works, it's fine.